src/rust/lib/graph-editor/src/component/node/connection.rs

```diff
@@ -0,0 +1,384 @@
+
+        Self {}
+    }
+}
+
+type ConnectionShapeView = component::ShapeView<ConnectionView>;
+
+/// Connections represent a link between an `InputPort` and an `OutputPort`.
+///
+/// Ports are shown as a connecting line going from one port to the other. The position snd shape
+/// of the `Connection` is defined by the start and end ports.
+///
+/// Connections can be created in the IDE by dragging from one port to another. This is currently handled in
+/// the `GraphEditor`, where events are collected and routed to allow dragging from ports,
+/// existence of a partially connected in-creation `Connection`, and finally the connecting of two
+/// ports.
+///
+/// TODO: use two line segments. This reduces emtpy space and allows partial glow.
+#[derive(CloneRef,Debug,Derivative)]
+#[derivative(Clone(bound=""))]
+#[allow(missing_docs)]
+pub struct Connection {
+    pub data : Rc<ConnectionData>
+}
+
+/// Internal data of `Connection`.
+#[derive(Debug,Clone)]
+#[allow(missing_docs)]
+pub struct ConnectionData {
+    start_port : RefCell<Option<WeakPort<OutputPortView>>>,
+    /// Start of the connection.
+    start      : Cell<Vector3<f32>>,
+    end_port         : RefCell<Option<WeakPort<InputPortView>>>,
+    /// End of the connection.
+    end        : Cell<Vector3<f32>>,
+
+    pub logger : Logger,
+    pub events : Events,
+    pub view   : Rc<ConnectionShapeView>,
+    pub label  : frp::Source<String>,
+}
+
+impl Connection {
+    /// Constructor.
+    pub fn new() -> Self {
+        frp::new_network! { connection_network
+            def label       = source::<String> ();
+            def port_move   = source::<()>     ();
+            def hover_start = source::<()>     ();
+            def hover_end   = source::<()>     ();
+        }
+        let network      = connection_network;
+        let start        = Cell::new(Vector3::zero());
+        let end          = Cell::new(Vector3::zero());
+        let logger       = Logger::new("connection");
+        let view         = Rc::new(component::ShapeView::new(&logger));
+        let events       = Events{network,port_move,hover_start,hover_end};
+        let start_port   = RefCell::new(None);
+        let end_port     = RefCell::new(None);
+
+        let data         = ConnectionData { start_port,start,end_port,end,label,events,view,
+                                            logger };
+        let data         = Rc::new(data);
+        Self {data} . init() .init_frp()
+    }
+
+    fn init(self) -> Self {
+        self
+    }
+
+    /// Set up the event handling for connections.
+    ///
+    /// Specifically:
+    /// * translates mouse events in hover events.
+    /// * sets up actions on port movement
+    /// * sets up actions on hover events
+    fn init_frp(self) -> Self {
+        let weak_connection = self.downgrade();
+        let network = &self.data.view.events.network;
+        frp::new_bridge_network! { [network,self.data.events.network]
```

Comment:
Its late so Im may be missing something, but why do you need a bridge network here?

Reply:
Actually I think I don't. Has been changed to `extend`.